06week/Checkpoint2.js

```diff
@@ -0,0 +1,45 @@
+
+// Given 1000 digits of PI as strings, return an array of the digits as numbers
+const nums = strNums.join();
+
+console.log(nums);
+
+// Find the sum of the even values
+//had to filter even numbers first
+const findEvens = strNums.filter(num => num % 2 ===0);
+//adding the total of the even values through reduce
+const sumEvens = findEvens.reduce((total, num) => {
+  return [(+total) + (+num)];
+});
+
+console.log(sumEvens);
+
+// Return an object with counts for each number
+// {
+//   0: 0, <== increment these counts
+//   1: 0,
+//   2: 0,
+//   ...
+//   8: 0,
+//   9: 0
+// }
+
+//return an object after manipulating the strNums array,
+//for each is indicated, so I used that function
+const numCount = {};
+nums.forEach((num) => {
+  numCount[num] = nums.filter(x => x === num).length;
```

Comment:
this is overly complicated, why are you running a filter on every number? this means for every number you are looping through the whole array to search for something unnecessary. The final object should look something like this `{0:1, 1:4, 2:1, 3:5}` where the key is the index and the value is the number in the array. 

Answer `x.forEach((num, index) =>{
   numCount[index] = num
})`